File: 백준/Silver/11047. 동전 0/동전 0.py
import sys
input = sys.stdin.readline

n, k = map(int,input().split())

coins = []
for i in range(n):
    coin = int(input())
    if coin <= k:
        coins.append(coin)
cnt = 0
for i in coins[::-1]:
    cnt += k // i
    k = k % i
    if k == 0:
        break
print(cnt)

File: 백준/Silver/11047. 동전 0/동전 0.swift
import Foundation

let input = readLine()!.split(separator: " ").map { Int($0)! }
let (n, k) = (input[0], input[1])
var count = 0
var sum = 0
var money = [Int]()
for _ in 0..<n {
    let num = Int(readLine()!)!
    money.append(num)
}
money.sort(by: >)
for i in 0..<n {
    if money[i] > k { continue }
    while true {
        if sum + money[i] > k { break }
        sum += money[i]
        count += 1
    }
}
print(count)